### DISCUSSION TITLE
zoom level biais

### DISCUSSION THREAD

**Discussion by farfromrefug** [maplibre/maplibre-gl-js#4818] [Q&A]
I want like to render the map with a zoom level biais.
By that i mean that i would like maplibre to render as if it was at zoom+biais to render a lot more details at a specific level.
I know it would require a lot more resources but it is what i need for my usecase.
My use case is to print a high dpi map in a3. But it also means i can render more details because it will be printed on paper.
Zoom level biais is the way to go.
I tried to find a way to do that in maplibre but could not find it. Is there any way to do it?
I am interested even if i need to modify the code of maplibre.

Thanks

Here is an example of what zoom level biais looks like on mobile with a framework which supports it
![Screenshot_20241010-160356_Alpi_Maps](https://github.com/user-attachments/assets/077a263c-fa8d-4a38-ba06-cb876ba38128)
![Screenshot_20241010-160418_Alpi_Maps](https://github.com/user-attachments/assets/7a32384d-b154-4f04-bc3c-14e8cbed38a2)

**Comment by HarelM** [comment]
You can add more feature to the tiles and define the style to show them on lower zoom level, can't you? 
You can also use addProtocol to load more tiles and merge them together and feed that to the renderer, I believe.

**Reply by farfromrefug** [reply]
@HarelM it would mean re creating the tiles but also changing the style. Zoom level biais allows the same render without having compute tiles)change style. Very practical to like print maps in a3 format or as posters

**Reply by HarelM** [reply]
Try the add protocol then...

**Reply by farfromrefug** [reply]
@HarelM what does that mean?

**Reply by HarelM** [reply]
https://maplibre.org/maplibre-gl-js/docs/API/functions/addProtocol/

**Reply by farfromrefug** [reply]
@HarelM thanks but I dont understand. How can that help me with what I am trying to achieve ? I understand what the protocol is for but this is not what I want. 
What I need is a way to have maplibre to render tiles from higher zoom level. What I mean is let's say maplibre is supposed to render xy tile at zoom z, i want instead to render the 4 tiles of zoom z+1 which correspond to xy. 
And keep on doing that if the biais is bigger. 
It Will.use far more CPU/GPU bût inam Ok with that.

I am sûre this is possible by modifying the code somehow but I dont know the maplibre JS code base which is why I am asking here

**Comment by zerda-ocm** [comment]
I am not, sure if this is what you want to achieve, but maybe it is an option to use CSS?

`#map {
      width: 200%;
      height:200%;
      scale:50%;
}`

You still have to adjust the zoom level, but you can render more tiles per area.

#map { width: 100%; height:100%; scale:100%; }:
![scale1](https://github.com/user-attachments/assets/9d1fbc6b-9d3c-44dc-8be6-5f11ba8c5ba8)

#map { width: 200%; height:200%; scale:50%; }:
![scale2](https://github.com/user-attachments/assets/7d9099f2-9c04-43dd-a8e4-0e235c9647f8)

**Reply by farfromrefug** [reply]
@zerda-ocm it seems to be exactly what I want ! Awesome 
Will try today and mark as answered if it works. 
Thank you!!!

**Reply by zerda-ocm** [reply]
If this is solely for yourself/internal use and doesn't need to be published as a web application, you can have the same effect with the help of the developer tools of most browsers:
Ctrl+Shift+I to start the developer console
Ctrl+Shift+M to simulate different resolutions (just double your normal resolution and scale it down if neccessary).
Chrome/Edge workes best for me and you can easily download the image as well.

**Reply by farfromrefug** [reply]
@zerda-ocm i can confirm it works! awesome solution. I had to add `left:-50%` and `right:-50%`
The only issue with that solution is style items "scale". as we scale the canvas, all fonts and rendered features are scaled.

I will still go over maplibre base code to see if there is a better solution

Thanks again

**Reply by maximumsomething** [reply]
@farfromrefug Did you ever find a better solution?

**Reply by HarelM** [reply]
For printing I think you might be able to use a similar script to the one used for generating images for the examples. 
This script opens the browser and takes a screen shot, if you write it so that it opens side by side images and later on stitch them together you might be able to achieve similar results, I think, but the browser resolution is probably the best solution.

**Reply by farfromrefug** [reply]
@maximumsomething no better solution on the web. Easily done on mobile sdk but it seems it was never thought to be something interesting in web. The width/left/right trick works but it makes thing really "slow" as the rendering gets huge.
